From 39be8b277c666db21965d54d5b72418ce57a450b Mon Sep 17 00:00:00 2001
From: Michael Biebl <biebl@debian.org>
Date: Tue, 19 May 2015 23:38:11 +0200
Subject: [PATCH] Add support for libsystemd

In systemd v209 the various libsystemd-* libraries were merged into a
single libsystemd library. So first check for this library and fall back
to libsystemd-login if not found.

Bug: https://bugzilla.gnome.org/show_bug.cgi?id=749603
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=779766

Upstream-Status: Applied
---
 configure.ac | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index ee3475d..bd5c043 100644
--- a/configure.ac
+++ b/configure.ac
@@ -264,18 +264,20 @@ dnl **********************************
 dnl *** Check for libsystemd-login ***
 dnl **********************************
 
-AC_ARG_ENABLE(libsystemd_login, AS_HELP_STRING([--disable-libsystemd-login],[build without liblibsystemd-login]))
+AC_ARG_ENABLE(libsystemd_login, AS_HELP_STRING([--disable-libsystemd-login],[build without libsystemd-login]))
 msg_libsystemd_login=no
 LIBSYSTEMD_LOGIN_LIBS=
 LIBSYSTEMD_LOGIN_CFLAGS=
 LIBSYSTEMD_LOGIN_REQUIRED=44
 
 if test "x$enable_libsystemd_login" != "xno"; then
-  PKG_CHECK_EXISTS([libsystemd-login >= $LIBSYSTEMD_LOGIN_REQUIRED], msg_libsystemd_login=yes)
+  PKG_CHECK_EXISTS([libsystemd], [msg_libsystemd_login=yes],
+                   [PKG_CHECK_EXISTS([libsystemd-login >= $LIBSYSTEMD_LOGIN_REQUIRED], msg_libsystemd_login=yes)])
 
   if test "x$msg_libsystemd_login" = "xyes"; then
-    PKG_CHECK_MODULES([LIBSYSTEMD_LOGIN],[libsystemd-login >= $LIBSYSTEMD_LOGIN_REQUIRED])
-    AC_DEFINE(HAVE_LIBSYSTEMD_LOGIN, 1, [Define to 1 if liblibsystemd_login is available])
+    PKG_CHECK_MODULES([LIBSYSTEMD_LOGIN],[libsystemd],,
+                      [PKG_CHECK_MODULES([LIBSYSTEMD_LOGIN],[libsystemd-login >= $LIBSYSTEMD_LOGIN_REQUIRED])])
+    AC_DEFINE(HAVE_LIBSYSTEMD_LOGIN, 1, [Define to 1 if libsystemd_login is available])
   fi
 fi
 
-- 
2.5.0

